src/AssocArray.php

```diff
@@ -216,6 +217,32 @@ public function withoutElement($value): AssocArray
         return $this->filter(Filters::notEqual($value));
     }
 
+    /**
+     * @phpstan-return IterableValue<TKey, TValue>
+     */
+    public function toIterableValue(): IterableValue
+    {
+        return new InfiniteIterableValue($this->toAssocArray());
+    }
+
+    /**
+     * @phpstan-return AssocValue<int|string, TKey>
```

Comment:
It can't be `AssocValue<TValue, TKey>`?
I know it doesn't make sense if TValue isn't int|string, but `AssocValue<int, string>` flipped should be rather `AssocValue<string, int>` tham `AssocValue<int|string, int>`